### DISCUSSION TITLE
幾個討論：粉絲數，出走勝利，育成方向

### DISCUSSION THREAD

**Discussion by Splendent** [NateScarlet/auto-derby#61] [General]
在commit 5aba858fddc2f8e59836fb2d731c125818fdfcc7 中加入的auto race功能
目前用了幾次發現幾個可以討論的地方
雖然以目前小樣本來說，養出來的評級似乎沒有差異太大，應該是在技能點跟基礎數值做了一定的交換
但在目前的雙子杯/長距離，或許可以考慮養單純的高數值馬 (目標耐速1200力900，雖然我的支援卡是不可能達到 💩 )
畢竟我也靠低智皇帝凹到勝利過 👑 

# 粉絲數

在養成東海帝王時發現他非常容易在初期去跑pre-op/op/g3等比賽
這在個人認知內是有些不划算的，初期應該要盡量訓練堆高支援卡的友情值？

在參考`race.py L:328 score function`後推測
`fan_score + prop + skill * 0.5 + not_winning_score - continuous_race_penalty`
比賽分數參考下列數值
1. 增加粉絲數
2. 數值
3. 技能點
4. 未出道分數(初次出賽失敗，隨回數增加 👍 )
5. 連續出走負面效果

## 討論：
個人認為或許可以降低粉絲數的權重；或者降低低級比賽的權重分數；又或者在初期就降低比賽分數的權重值

## 附件：
目前的小樣本比較，都是一萬出頭分
### 10,153
![image](https://user-images.githubusercontent.com/3349355/122243355-9f9bda00-cef6-11eb-9bad-a82b0d8fd44a.png)
### 11,302
![image](https://user-images.githubusercontent.com/3349355/122243869-06b98e80-cef7-11eb-9d8b-2bf503b2d3cb.png)
### 10,836
![image](https://user-images.githubusercontent.com/3349355/122243943-1638d780-cef7-11eb-86c5-ea3655551a77.png)

### 10,330
養成紀錄： [tokaiteio_pre-op.zip](https://github.com/NateScarlet/auto-derby/files/6664202/tokaiteio_pre-op.zip)
![image](https://user-images.githubusercontent.com/3349355/122243240-8abf4680-cef6-11eb-8494-5890ec215002.png)

# 出走勝利建議
`预估比赛结果，如果不能仅靠属性拿冠军则暂停请求人工确认`
在養成皇帝的時候，以我(奈米課)以及我朋友(小課)的幾個樣本中，皇帝幾乎在大多數的比賽都會出現人工確認提示
使得自動養成幾乎無法達成，出賽都需要手動確定

另外即使點了技能，但有可能發動條件不同，例如技能要差馬、或者繼承的固有技能需要追過的條件，出賽卻自動選了追/先
這部分沒有特別TRACE，不確定是如何運作的

## 討論：
或許在NURTURING的時候可以選擇要不要比賽提示，或者提供比賽名次容忍範圍

## 附件：
附上今天出廠的皇帝
### 10,094
![image](https://user-images.githubusercontent.com/3349355/122248297-914fbd00-cefa-11eb-981c-28bb0ddb939c.png)

# 養成偏向
這個又不得不提到雙子杯了，就目前的理解來說，雙子杯需要高速極高耐，因此前陣子在養成的時候有稍微修改了一下數值權種
在免強跑得過URA的情況下養出了一隻個人覺得過得去的皇帝，至少在雙子杯取勝的紀錄
或許可以考慮提供不同的養成PROFILE，但是這又考慮到每個人的養成觀念多少有差異，提供其他選擇搞不好也只是徒勞

增加工作量而已 🐒 

## 附件：
附上上文提到的皇帝 
沒記錯的話是直接把速度跟耐力的判斷數值互相交換 🏇 
### 10,397
![image](https://user-images.githubusercontent.com/3349355/122250162-04a5fe80-cefc-11eb-9704-68c387ab19fc.png)

**Comment by NateScarlet** [comment]
现在的比赛评分仅仅是简单数值比例相加
我正在改成不仅是而是考虑力量不达标被阻挡 耐力不达标失速等情况
今天一直在调

开发中的方案是
90% 概率第一名为 10000 分
90% 概率排名正中间为 5000 分

根据速度确定基本分为 0-15000 分  然后其他不利条件在这上面减分数

**Comment by NateScarlet** [comment]
![图片](https://user-images.githubusercontent.com/21083014/122267228-3a9fae80-cf0d-11eb-8489-7b04ed3d9b44.png)
加了暂停条件设置 0bb3e61

**Reply by Splendent** [reply]
除了動作真快我也不知道該說啥了 👍

**Reply by fatinghenji** [reply]
race order指的是预估比赛排名吗？

**Reply by NateScarlet** [reply]
> 
> 
> race order指的是预估比赛排名吗？

对

**Comment by NateScarlet** [comment]
参考绿帽评价调整了一晚的数据 现在预测水平应该和绿帽差不多

**Reply by fatinghenji** [reply]
辛苦！

**Comment by NateScarlet** [comment]
关于育成配置文件

尝试添加了插件支持，用户可以这样自定义：

https://github.com/NateScarlet/auto-derby/blob/a448f221acb7c6fdaaeed7de52f8197e2032624f/plugins/example_custom_race_score.py#L5-L10

**Reply by fatinghenji** [reply]
没太看懂代码，是调整某一场比赛的优先度吗？

**Reply by NateScarlet** [reply]
> 
> 
> 没太看懂代码，是调整某一场比赛的优先度吗？

对 给有马纪念加10分 1分相当于300速时的1速

**Reply by Splendent** [reply]
可惜我沒法再給第二顆星星 ⭐ 
fork測試中.... 🏇
